File: playwright-python-e2e-template/pages/login_page.py
from pages.base_page import BasePage
from config import Config

class LoginPage(BasePage):
    def __init__(self, page):
        super().__init__(page)
        self.username_input = "input[type=email], input[name=username]"
        self.password_input = "input[type=password]"
        self.submit_button = "button[type=submit]"

    async def open(self):
        await self.navigate(Config.LOGIN_PATH)

    async def login(self, username, password):
        await self.page.fill(self.username_input, username)
        await self.page.fill(self.password_input, password)
        await self.page.click(self.submit_button)

File: playwright-java-e2e-template/src/main/java/pages/LoginPage.java
package pages;

import com.microsoft.playwright.Page;

public class LoginPage extends BasePage {
    private static final String USERNAME_SELECTOR = "[data-testid='username'], #username, input[name='username']";
    private static final String PASSWORD_SELECTOR = "[data-testid='password'], #password, input[name='password']";
    private static final String SUBMIT_SELECTOR = "[data-testid='submit'], button[type='submit'], input[type='submit']";

    public LoginPage(Page page) {
        super(page);
    }

    public void open() {
        navigate("/login");
    }

    public void login(String username, String password) {
        page.fill(USERNAME_SELECTOR, username);
        page.fill(PASSWORD_SELECTOR, password);
        page.click(SUBMIT_SELECTOR);
    }
}